### ISSUE TITLE
Improve our bandwidth usage strategy

### ISSUE THREAD

**Issue by bassosimone** [measurement-kit/measurement-kit#1378]
People has complained that ooniprobe mobile apps use too much bandwidth. It was not totally clear whether this was related to the overall amount of data consumed or to the instantaneous speed.

Regardless, I'm taking this input as a starting point to elaborate more on our bandwidth usage strategy.

There are already other open tickets on that and later we can probably expand / merge.

Things that according to me are relevant:

- [ ] allow users to tweak the parallelism of running measurements so that they can scale down the bandwidth used by running e.g. web_connectivity

- [ ] implement a mechanism to submit results when we're not measuring, both to avoid fucking up measurements because we're uploading and generally to reduce instantaneous used bandwidth

- [ ] (as probably mentioned in another ticket) implement zlib so to compress uploaded reports; see #1315 

- [ ] (as probably mentioned in another ticket) add support for measuring the amount of data consumed by each test template (I guess the test template is the correct place where to do that); see #1463 

- [ ] add rules for stopping measurements not only after a certain amount of time has passed but also when a specific amount of data has been received or sent

**Comment by bassosimone** [comment]
The most interesting bit of this issue has been moved at https://github.com/ooni/probe-engine/issues/85. The rest can be happily and merrily closed.
